bin/stringmerge

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+'''
+Merge two existing problems.
+'''
+
+import sys
+import argparse
+
+from stringfuzz.ast import ExpressionNode, SortNode, IdentifierNode, FunctionDeclarationNode, SortedVarNode
+from stringfuzz.ast_walker import ASTWalker
+from stringfuzz.constants import LANGUAGES, SMT_25_STRING
+from stringfuzz.generator import generate
+from stringfuzz.parser import parse, ParsingError
+from stringfuzz.smt import smt_string_logic, smt_sat
+
+# defaults
+DEFAULT_RENAME_IDS = False
+
+# constants
+DECLARE_CONST = "declare-const"
+
+# helper functions 
+def assertion(expr):
+    if isinstance(expr, ExpressionNode):
+        symbol = expr.symbol
+        if isinstance(symbol, IdentifierNode) and symbol.name == "assert":
+            return True
+    return False
+
+def declaration(expr):
+    if isinstance(expr, (FunctionDeclarationNode, SortedVarNode)):
+        return True
+    if isinstance(expr, ExpressionNode):
+        symbol = expr.symbol
+        if isinstance(symbol, IdentifierNode) and symbol.name == DECLARE_CONST:
+            return True
+    return False
+
+def sort(expr):
+    return isinstance(expr, SortNode)
+
+# helper class
+class RenameIDWalker(ASTWalker):
+    def __init__(self, ast, suffix):
+        super(RenameIDWalker, self).__init__(ast)
+        self.suffix = str(suffix)
+
+    def exit_identifier(self, identifier, parent):
+        identifier.name += self.suffix
+
+# most important function
+def merge(asts, rename_ids):
+    if rename_ids:
+        for i in range(len(asts)):
+            asts[i] = RenameIDWalker(asts[i], i).walk()
```

Comment:
In the case of numeric variable names (e.g. `var0`), this might actually still result in *seemingly* colliding names (e.g. `var1` -> `var10`. I did the mental exercise and it seems like they still won't collide, but maybe change the prefix to be something like `"{}_".format(i)` instead, so it's easier to see that they've been renamed?

Reply:
Sure! They won't collide, but the change in name will make that more clear